### ISSUE TITLE
[java] SimplifyConditional: False negative when null check and instanceof are separated by other && conditions

### ISSUE THREAD

**Issue by leemeii** [pmd/pmd#6513]
**Affects PMD Version:** 7.22.0

**Rule:**[SimplifyConditional](https://pmd.github.io/pmd/pmd_rules_java_design.html#simplifyconditional)

**Description:**
The **SimplifyConditional** rule only detects the pattern `x != null && x instanceof Type` when the null check is **directly adjacent** to the instanceof expression. 
When additional conditions are inserted in the && chain, the rule fails to trigger even though the null check before instanceof remains completely redundant (semantically identical after removal).
This is very **common** in real-world code, where developers frequently **insert additional non-null checks or other conditions in the middle of the && chain** .

**Code Sample demonstrating the issue:**

```java
public class a{
    public void foo(){
        String s = "a";
        if( s!=null&&s instanceof String){   // Scenario A: Direct adjacent case (correctly reported - TP)
            System.out.println(s);
        }

        String s2="a";
        if(s!=null && s2!=null && s instanceof String){   // Scenario B: Intervening condition (False Negative)
            System.out.println(s);
        }
    }
}
```

**Expected outcome:**

PMD should report a violation at line 10 with the message:
"No need to check for null before an instanceof"
But currently it only reports the first if, completely missing the second one. This is a false-negative.

**Running PMD through:** *CLI*
